### ISSUE TITLE
Exclude extensions for Jython

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3410]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#3410|
|Type|enhancement|
|Created|2008-09-05 15:34:46Z|
|Branch|https://github.com/twisted/twisted/tree/jython-ext-3410|

When trying to build Twisted for Jython it dies with:

```
$ jython setup.py install
running install
running build
running build_py
copying twisted/protocols/dns.py -> build/lib.java1.6.0_05-2.3/twisted/protocols
copying twisted/protocols/ethernet.py -> build/lib.java1.6.0_05-2.3/twisted/protocols
copying twisted/protocols/imap4.py -> build/lib.java1.6.0_05-2.3/twisted/protocols
copying twisted/protocols/ip.py -> build/lib.java1.6.0_05-2.3/twisted/protocols
copying twisted/protocols/irc.py -> build/lib.java1.6.0_05-2.3/twisted/protocols
copying twisted/protocols/jabber.py -> build/lib.java1.6.0_05-2.3/twisted/protocols
copying twisted/protocols/msn.py -> build/lib.java1.6.0_05-2.3/twisted/protocols
copying twisted/protocols/nntp.py -> build/lib.java1.6.0_05-2.3/twisted/protocols
copying twisted/protocols/oscar.py -> build/lib.java1.6.0_05-2.3/twisted/protocols
copying twisted/protocols/pop3.py -> build/lib.java1.6.0_05-2.3/twisted/protocols
copying twisted/protocols/raw.py -> build/lib.java1.6.0_05-2.3/twisted/protocols
copying twisted/protocols/rawudp.py -> build/lib.java1.6.0_05-2.3/twisted/protocols
copying twisted/protocols/smtp.py -> build/lib.java1.6.0_05-2.3/twisted/protocols
copying twisted/protocols/sux.py -> build/lib.java1.6.0_05-2.3/twisted/protocols
copying twisted/protocols/toc.py -> build/lib.java1.6.0_05-2.3/twisted/protocols
copying twisted/web/xmlrpc.py -> build/lib.java1.6.0_05-2.3/twisted/web
copying twisted/web/test/test_xmlrpc.py -> build/lib.java1.6.0_05-2.3/twisted/web/test
running build_ext
error: Compiling extensions is not supported on Jython
```

jbaker suggested to use something like:

```python
if not sys.platform.startswith('java'):
    # ...
```

<details><summary>Searchable metadata</summary>

```
trac-id__3410 3410
type__enhancement enhancement
reporter__thijs thijs
priority__normal normal
milestone__ 
branch__branches_jython_ext_3410 branches/jython-ext-3410
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__jython jython
time__1220628886000000 1220628886000000
changetime__1220734548000000 1220734548000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_1"></a>|@thijstriemstra commented|
|-|-|

The command for the Jython buildslave also needs to change and exclude the extension build process.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

Replying to [thijs](#note_1):
> The command for the Jython buildslave also needs to change and exclude the extension build process.

Cause it currently dies with http://buildbot.twistedmatrix.com/builders/ubuntu64-jython2.5-select/builds/5/steps/compile/logs/stdio/text

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

(In [24687]) Branching to 'jython-ext-3410'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra removed owner|
|-|-|

Replying to [thijs](#note_1):
> The command for the Jython buildslave also needs to change and exclude the extension build process.

I'll open a new ticket for this as suggested by exarkun.

I committed a fix in r24688, and the buildslave now [shows](http://buildbot.twistedmatrix.com/builders/ubuntu64-jython2.5-select/builds/6/steps/compile/logs/stdio/text) the expected output:

```
running build_ext
```

Putting it up for review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

I had some ideas for doing this check in `twisted/python/dist.py` where it could be unit tested.  However, it seems that `ConditionalExtension`, the thing I was hoping we would be able to use, does not actually work unless you can run `build_ext`.  As it seems Jython can't do this, I suppose there's no obvious way to use `twisted/python/dist.py` to do this.

So, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [24692]) Merge jython-ext-3410: exclude extensions in the build process when using Jython

Author: thijs
Reviewer: exarkun
Fixes: [#3410](https://github.com/twisted/twisted/issues/3410)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
